hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/MetricsWAL.java

```diff
@@ -58,9 +59,10 @@ public void postSync(final long timeInNanos, final int handlerSyncs) {
   @Override
   public void postAppend(final long size, final long time, final WALKey logkey,
       final WALEdit logEdit) throws IOException {
-    source.incrementAppendCount();
+    TableName tableName = logkey.getTableName();
```

Comment:
Is it legal for an implementation to pass in a null WALKey, or a non-null WALKey with a null TableName? (As was done in the prior version of TestMetricsWAL?) If so, we may need some NPE protection.

Reply:
Don't think it's optional (the test I fixed seems like an outlier). At least the protobuf based WAL implementation mandates the table name to be a required field and there are multiple other places in the code (like replication tailing etc) that assume a non-null table name. 

```
message WALKey {
  required bytes encoded_region_name = 1;
  required bytes table_name = 2;
```

Wish there is a better way to enforce it but using (non)nullable annotations is not a common practice in HBase code.